js/custom_projection.js

```diff
@@ -1,146 +1,290 @@
+// http://egsc.usgs.gov/isb/pubs/MapProjections/projections.html#albers
+function albersUsaTerritories() {
+  var cache,
+      cacheStream,
+      lower48 = d3.geoAlbers(), lower48Point,
+      alaska = d3.geoConicEqualArea().rotate([154, 0]).center([-2, 58.5]).parallels([55, 65]), alaskaPoint, // EPSG:3338
+      hawaii = d3.geoConicEqualArea().rotate([157, 0]).center([-3, 19.9]).parallels([8, 18]), hawaiiPoint, // ESRI:102007
+      puertoRico = d3.geoConicEqualArea().rotate([66, 0]).center([0, 18]).parallels([8, 18]), puertoRicoPoint, //Taken from https://bl.ocks.org/mbostock/5629120
+      point, pointStream = {point: function(x, y) { point = [x, y]; }};
+
+      /*
+      var puertoRicoBbox = [[-68.3, 19], [-63.9, 17]];
+      */
 
   function albersUsa(coordinates) {
     var x = coordinates[0], y = coordinates[1];
-    point = null;
-    (lower48Point(x, y), point)
-        || (alaskaPoint(x, y), point)
-        || (hawaiiPoint(x, y), point)
-        || (puertoRicoPoint(x, y), point);
-    return point;
+
+    return point = null, (lower48Point.point(x, y), point) ||
+        (alaskaPoint.point(x, y), point)  ||
+        (hawaiiPoint.point(x, y), point)  ||
+        (puertoRicoPoint.point(x, y), point);
   }
 
   albersUsa.invert = function(coordinates) {
+
     var k = lower48.scale(),
         t = lower48.translate(),
         x = (coordinates[0] - t[0]) / k,
         y = (coordinates[1] - t[1]) / k;
-    return (y >= .120 && y < .234 && x >= -.425 && x < -.214 ? alaska
-        : y >= .166 && y < .234 && x >= -.214 && x < -.115 ? hawaii
-        : y >= .204 && y < .234 && x >= .320 && x < .380 ? puertoRico
+        /*
+        //How are the return values calculated:
+        console.info("******");
+        var c0 = puertoRico(puertoRicoBbox[0]);
+        var x0 = (c0[0] - t[0]) / k;
+        var y0 = (c0[1] - t[1]) / k;
+        console.info("p0 puertoRico", x0 + ' - ' + y0);
+        var c1 = puertoRico(puertoRicoBbox[1]);
+        var x1 = (c1[0] - t[0]) / k;
+        var y1 = (c1[1] - t[1]) / k;
+        console.info("p1 puertoRico", x1 + ' - ' + y1);
+        c0 = samoa(samoaBbox[0]);
+        x0 = (c0[0] - t[0]) / k;
+        y0 = (c0[1] - t[1]) / k;
+        console.info("p0 samoa", x0 + ' - ' + y0);
+        c1 = samoa(samoaBbox[1]);
+        x1 = (c1[0] - t[0]) / k;
+        y1 = (c1[1] - t[1]) / k;
+        console.info("p1 samoa", x1 + ' - ' + y1);
+        c0 = guam(guamBbox[0]);
+        x0 = (c0[0] - t[0]) / k;
+        y0 = (c0[1] - t[1]) / k;
+        console.info("p0 guam", x0 + ' - ' + y0);
+        c1 = guam(guamBbox[1]);
+        x1 = (c1[0] - t[0]) / k;
+        y1 = (c1[1] - t[1]) / k;
+        console.info("p1 guam", x1 + ' - ' + y1);
+        */
+
+    return (y >= 0.120 && y < 0.234 && x >= -0.425 && x < -0.214 ? alaska
+        : y >= 0.166 && y < 0.234 && x >= -0.214 && x < -0.115 ? hawaii
+        : y >= 0.2064 && y < 0.2413 && x >= 0.312 && x < 0.385 ? puertoRico
```

Comment:
are PR & VI handled together?

Reply:
So that is what I need the actual shapes for, I can adjust and do what we need to get both, but until then I am just guessing.

Reply:
But I feel 95% sure on positivity they will be

Reply:
cool